command/flags.go

```diff
@@ -113,6 +111,26 @@ var InitFlags = []cli.Flag{
 	},
 }
 
+var RegisterFlags = []cli.Flag{
+	&cli.StringFlag{
+		Name:     "config",
+		Usage:    "Config file containing provider's peer ID and private key",
```

Comment:
why is the providers _private_ key needed?

Reply:
The register command is run by someone wanting to be a provider, and the request to register needs to be signed by the provider who is registering themselves and the indexer may be on a remote machine.  The provider being registered must be trusted by the indexer (indexer already knows indexer peerID), and the indexer needs to verify the request comes from the provider which the message signature does.  Trust just means that a provider can be registered without having to verify on-chain.